eos/form-factors/parametric-kkrvd2024.cc

```diff
@@ -0,0 +1,371 @@
+        double sum2 = 0.0;
+        for (auto i = 0u; i < b.size(); i++)
+        {
+            sum2 += b[i] * i * std::pow(z0, i-1);
+        };
+
+        const double xprime_z0 = ( 2.0 * (std::real(zr) - z0) * phitilde_z0 - std::norm(z0 - zr) * phitildeprime_z0 ) / power_of<2>(std::norm(z0 - zr) * phitilde_z0);
+
+        const double dFdz_z0 = sum1 * xprime_z0 + sum2 / (phitilde_z0 * std::norm(z0 - zr)); // the weight factor has been absorbed into 1 / phi
+        return dFdz_z0 * std::real(this->dzdq2(0.0));
+    }
+
+    double
+    KKRvD2024FormFactors<VacuumToPiPi>::r_pi_squared() const
+    {
+        return 6.0 * this->dFdq2_q2eq0() * power_of<2>(this->_hbarc());
+    }
+
+    double
+    KKRvD2024FormFactors<VacuumToPiPi>::r_pi() const
+    {
+        return sqrt(6.0 * this->dFdq2_q2eq0()) * this->_hbarc();
+    }
+
+    complex<double> KKRvD2024FormFactors<VacuumToPiPi>::residue_rho() const
+    {
+        // Super-threshold pole location
+        const auto zr       =  this->_zr(this->_M_fp_I1(), this->_G_fp_I1());
+        const auto chi      = 0.0258815; // GeV^-2, from Meril at Q^2 = 1 GeV^2
+        const auto phitilde = this->phitilde_p(zr, chi);
+
+        // prepare expansion coefficients
+        std::array<double, 10> b;
+        std::copy(_b_fp_I1.cbegin(), _b_fp_I1.cend(), b.begin()+2);
+        // Fix b[0] and b[1] to enforce F(q2=0) = 1 and F'(q2=t+) = 0
+        b[0] = _b0_fp_I1(chi, zr);
+        b[1] = _b1_fp_I1(chi, zr);
+
+        const auto series = this->series_m(zr, b);
+
+        return series / (zr - std::conj(zr)) /  phitilde;
+    }
+
+    double KKRvD2024FormFactors<VacuumToPiPi>::re_residue_rho() const
+    {
+        return std::real(this->residue_rho());
+    }
+
+    double KKRvD2024FormFactors<VacuumToPiPi>::im_residue_rho() const
+    {
+        return std::imag(this->residue_rho());
+    }
+
+    double KKRvD2024FormFactors<VacuumToPiPi>::dispersive_integrand(const double & alpha) const
+    {
+        const complex<double> z = std::polar(1.0, alpha);
+        const complex<double> w = this->w(z);
+
+        const auto chi = 0.0258815; // GeV^-2, from Meril at Q^2 = 1 GeV^2
+
+        // Super-threshold pole location
+        const auto zr =  this->_zr(this->_M_fp_I1(), this->_G_fp_I1());
+
+        // prepare expansion coefficients
+        std::array<double, 10> b;
+        std::copy(_b_fp_I1.cbegin(), _b_fp_I1.cend(), b.begin()+2);
+        // Fix b[0] and b[1] to enforce F(q2=0) = 1 and F'(q2=t+) = 0
+        b[0] = _b0_fp_I1(chi, zr);
+        b[1] = _b1_fp_I1(chi, zr);
+
+        const auto series = this->series_m(z, b);
+
+        return std::norm(w * series / (z - zr) / (z - std::conj(zr)));
+    }
+
+    double KKRvD2024FormFactors<VacuumToPiPi>::saturation() const
+    {
+        std::function<double (const double &)> f = [this](const double & alpha) -> double { return this->dispersive_integrand(alpha); };
+        return integrate1D(f, 512, -M_PI, M_PI) / (2.0 * M_PI);
```

Comment:
Can we replace this with ``integrate<GSL::QNG>``? I would like to keep the number of users of ``integrate1D`` as low as possible.